Compile error with json symbol on frontera (@ TACC)

Thank you for taking the time to submit an issue!

## Background information

Not seeing this issue anywhere else.

### What version of the PMIx Reference Library are you using? (e.g., v1.0, v2.1, git master @ hash, etc.)

github master @ 949bab5f13759ba3e158ec389a2bd0041c06d78a

### Describe how PMIx was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

clone

### Please describe the system on which you are running

* Operating system/version: CentOS 7.6
* Computer hardware: x86
* Network type: IB

-----------------------------

## Details of the problem

```
make[2]: Entering directory `/home1/01858/arcurtis/src/pmix/build/src/mca/pnet/sshot'
depbase=`echo pnet_sshot.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/sh ../../../../libtool  --tag=CC   --mode=compile gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I../../../../../pmix-git/src/mca/pnet/sshot -I../../../../src/include -I../../../../include   -iquote/home1/01858/arcurtis/src/pmix/build -iquote/home1/01858/arcurtis/src/pmix/pmix-git -iquote/home1/01858/arcurtis/src/pmix/pmix-git/src -iquote/home1/01858/arcurtis/src/pmix/build/include -I/home1/01858/arcurtis/src/pmix/pmix-git/include  -O3 -DNDEBUG -Wall -Wundef -Wno-long-long -Wsign-compare -Wmissing-prototypes -Wstrict-prototypes -Wcomment -pedantic -Werror-implicit-function-declaration -finline-functions -fno-strict-aliasing -mcx16 -pthread -fvisibility=hidden -MT pnet_sshot.lo -MD -MP -MF $depbase.Tpo -c -o pnet_sshot.lo ../../../../../pmix-git/src/mca/pnet/sshot/pnet_sshot.c &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  gcc -std=gnu99 -DHAVE_CONFIG_H -I. -I../../../../../pmix-git/src/mca/pnet/sshot -I../../../../src/include -I../../../../include -iquote/home1/01858/arcurtis/src/pmix/build -iquote/home1/01858/arcurtis/src/pmix/pmix-git -iquote/home1/01858/arcurtis/src/pmix/pmix-git/src -iquote/home1/01858/arcurtis/src/pmix/build/include -I/home1/01858/arcurtis/src/pmix/pmix-git/include -O3 -DNDEBUG -Wall -Wundef -Wno-long-long -Wsign-compare -Wmissing-prototypes -Wstrict-prototypes -Wcomment -pedantic -Werror-implicit-function-declaration -finline-functions -fno-strict-aliasing -mcx16 -pthread -fvisibility=hidden -MT pnet_sshot.lo -MD -MP -MF .deps/pnet_sshot.Tpo -c ../../../../../pmix-git/src/mca/pnet/sshot/pnet_sshot.c  -fPIC -DPIC -o .libs/pnet_sshot.o
../../../../../pmix-git/src/mca/pnet/sshot/pnet_sshot.c: In function 'sshot_init':
../../../../../pmix-git/src/mca/pnet/sshot/pnet_sshot.c:210:24: error: implicit declaration of function 'json_error_code' [-Werror=implicit-function-declaration]
                        json_error_code(&error));
                        ^
../../../../../pmix-git/src/mca/pnet/sshot/pnet_sshot.c: In function 'allocate':
../../../../../pmix-git/src/mca/pnet/sshot/pnet_sshot.c:395:19: warning: variable 'coords' set but not used [-Wunused-but-set-variable]
     pmix_coord_t *coords;
                   ^
cc1: some warnings being treated as errors
make[2]: *** [pnet_sshot.lo] Error 1
make[2]: Leaving directory `/home1/01858/arcurtis/src/pmix/build/src/mca/pnet/sshot'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home1/01858/arcurtis/src/pmix/build/src'
make: *** [all-recursive] Error 1
```

--- a/config/pmix_check_jansson.m4
+++ b/config/pmix_check_jansson.m4
@@ -65,14 +65,30 @@ AC_DEFUN([PMIX_CHECK_JANSSON],[
     	pmix_check_jansson_save_LIBS="$LIBS"
 
         PMIX_CHECK_PACKAGE([pmix_check_jansson],
-		   [jansson.h],
-	 	   [jansson],
-		   [json_loads],
-		   [],
-		   [$pmix_check_jansson_dir],
-		   [$pmix_check_jansson_libdir],
-		   [pmix_check_jansson_happy="yes"],
-		   [pmix_check_jansson_happy="no"])
+		                   [jansson.h],
+	 	                   [jansson],
+		                   [json_loads],
+		                   [],
+		                   [$pmix_check_jansson_dir],
+		                   [$pmix_check_jansson_libdir],
+		                   [pmix_check_jansson_happy="yes"],
+		                   [pmix_check_jansson_happy="no"])
+
+        if test "$pmix_check_jansson_happy" == "yes"; then
+            AC_MSG_CHECKING([if libjansson version is 2.11 or greater])
+            AS_IF([test "$pmix_jansson_source" != "standard"],
+                  [PMIX_FLAGS_APPEND_UNIQ(CPPFLAGS, $pmix_check_jansson_CPPFLAGS)])
+            AC_COMPILE_IFELSE(
+                  [AC_LANG_PROGRAM([[#include <jansson.h>]],
+                  [[
+        #if JANSSON_VERSION_HEX < 0x00020b00
+        #error "jansson API version is less than 2.11"
+        #endif
+                  ]])],
+                  [AC_MSG_RESULT([yes])],
+                  [AC_MSG_RESULT([no])
+                   pmix_check_jansson_happy=no])
+        fi
 
     	CPPFLAGS="$pmix_check_jansson_save_CPPFLAGS"
     	LDFLAGS="$pmix_check_jansson_save_LDFLAGS"

